patches/chromium/feat_corner_smoothing_css_rule_and_blink_painting.patch

```diff
@@ -268,28 +268,28 @@ index f4cdee12ea4352067f5de3e074e43d51ef56d2e5..6377e4b1ea8aa46b0bf69f8420b6c439
    ],
  }
 diff --git a/third_party/blink/renderer/core/paint/box_painter_base.cc b/third_party/blink/renderer/core/paint/box_painter_base.cc
-index 57f71cbd80d88416693e5b83f695fd02eb7a405b..d638da7a734fa40de118d056c59488b63caff7e2 100644
+index 71e68a3bfc83869d44da560e3eb151a8baced676..c9e88fa2178c4cafb61cd7c75272455f5396a872 100644
 --- a/third_party/blink/renderer/core/paint/box_painter_base.cc
 +++ b/third_party/blink/renderer/core/paint/box_painter_base.cc
-@@ -417,16 +417,20 @@ void BoxPainterBase::PaintInsetBoxShadow(const PaintInfo& info,
-     ContouredRect inner_rounded_rect(
-         FloatRoundedRect(inner_rect, bounds.GetRadii()));
-     ApplySpreadToShadowShape(inner_rounded_rect, -shadow.Spread());
+@@ -412,16 +412,20 @@ void BoxPainterBase::PaintInsetBoxShadow(const PaintInfo& info,
+     ContouredRect inner_contoured_rect(
+         FloatRoundedRect(inner_rect, bounds.GetRadii()),
+         bounds.GetCornerCurvature());
 +    ContouredRect contoured_bounds(
-+        bounds, ContouredBorderGeometry::ContouredBorder(
++        bounds.AsRoundedRect(), ContouredBorderGeometry::ContouredBorder(
 +                    style, PhysicalRect::EnclosingRect(bounds.Rect()))
 +                    .GetCornerCurvature());
-     if (inner_rounded_rect.IsEmpty()) {
+     ApplySpreadToShadowShape(inner_contoured_rect, -shadow.Spread());
+     if (inner_contoured_rect.IsEmpty()) {
        // |AutoDarkMode::Disabled()| is used because |shadow_color| has already
        // been adjusted for dark mode.
--      context.FillRoundedRect(bounds, shadow_color, AutoDarkMode::Disabled());
+-      context.FillContouredRect(bounds, shadow_color, AutoDarkMode::Disabled());
 +      context.FillContouredRect(contoured_bounds, shadow_color, AutoDarkMode::Disabled());
        continue;
      }
      GraphicsContextStateSaver state_saver(context);
      if (bounds.IsRounded()) {
-       // TODO(crbug.com/397459628) render corner-shape with box-shadow
--      context.ClipContouredRect(ContouredRect(bounds));
+-      context.ClipContouredRect(bounds);
 +      context.ClipContouredRect(contoured_bounds);
```

Comment:
This section of the patch can be removed now! 😄 The upstream code should be fine as-is.

Reply:
Thanks for the info!

Removed in 701318be955daee8b1d0d5fbfc9fd0d64ed06d37